File: PUGD-frontend/graphql/mutations/acquisition/facture.js
import gql from "graphql-tag";

const InsertFacture = gql`
  mutation(
    $id: String!
    $numFacture: String!
    $provider: String!
    $order: String!
    $status: String!
    $payementDate: String!
    $total_ttc: Float!
    $currency: String!
    $date: String!
    $establishement: String!
    $quantitiesFactured: [Float!]!
    $order_lines: [String!]!
  ) {
    insertFacture(
      id: $id
      provider: $provider
      order: $order
      status: $status
      establishement: $establishement
      numFacture: $numFacture
      payementDate: $payementDate
      total_ttc: $total_ttc
      currency: $currency
      date: $date
      order_lines: $order_lines
      quantitiesFactured: $quantitiesFactured
    )
  }
`;

const DeleteFacture = gql`
  mutation($_id: String!) {
    deleteFacture(_id: $_id)
  }
`;
module.exports = {
  InsertFacture,
  DeleteFacture,
};

File: PUGD-api/graphql/types/AcquisitionTypes/facture.go
package AcquisitionTypes

import "github.com/graphql-go/graphql"

var FactureType = graphql.NewObject(graphql.ObjectConfig{
	Name: "FactureType",
	Fields: graphql.Fields{
		"_id": &graphql.Field{
			Type: graphql.ID,
		},
		"numFacture": &graphql.Field{
			Type: graphql.String,
		},
		"provider": &graphql.Field{
			Type: graphql.String,
		},
		"order": &graphql.Field{
			Type: graphql.ID,
		},
		"status": &graphql.Field{
			Type: graphql.String,
		},
		"date": &graphql.Field{
			Type: graphql.String,
		},
		"payementDate": &graphql.Field{
			Type: graphql.String,
		},
		"total_ttc": &graphql.Field{
			Type: graphql.Float,
		},
		"currency": &graphql.Field{
			Type: graphql.String,
		},
		"establishement": &graphql.Field{
			Type: graphql.String,
		},

		"quantitiesFactured": &graphql.Field{
			Type: graphql.NewList(graphql.Float),
		},
		"order_lines": &graphql.Field{
			Type: graphql.NewList(graphql.ID),
		},
	},
})
